File: ios/PSMobile/Core/PrinterReadyPolicy.swift
import Foundation

/// Determines whether the workspace has a printer that can be used safely.
/// Kept independent from the slicer core so the rule can be tested without
/// loading the native engine.
enum PrinterReadyPolicy {
    static func isReady(selectedPrinter: String?, installedPrinters: Set<String>) -> Bool {
        guard let selectedPrinter,
              !selectedPrinter.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty else {
            return false
        }
        return installedPrinters.contains(selectedPrinter)
    }
}

File: android/app/src/main/java/de/psmobile/ui/PrinterReadyPolicy.kt
package de.psmobile.ui

/** The workspace is usable only with an installed, selected printer preset. */
object PrinterReadyPolicy {
    fun isReady(selectedPrinter: String, installedPrinters: List<String>): Boolean =
        selectedPrinter.isNotBlank() && selectedPrinter in installedPrinters
}
